### ISSUE TITLE
Fine tune network bandwidth utilization

### ISSUE THREAD

**Issue by baylanger** [erkexzcx/stoppropaganda#74]
When this app runs with 1 worker and 25 dns workers, network stats show ~ 65Mbps down / 6Mbps up.

With 25 workers & 25 dns workers, 150Mbps / 23Mbps - but values are above my plan (120/20!).

Can we introduce a way for end-user to adjust the introduction of a sleep in ms between each queries or any other way?  Many users in N. America are stuck low bandwidth pipes, as low as ~ 6Mbps/1Mbps.

[edit] this is with tons of sites currently returning a 300 redirect or 400 err.

**Comment by erkexzcx** [comment]
I have 2 ideas:
- Use pool of workers (e.g. 100 shared workers for all websites)
- Add flag to disable downloading the response content.

I will look into this tomorrow.

**Comment by 0xzheo** [comment]
I currently deal with this by limiting the CPU usage of the process using cpulimit.

```
#!/bin/bash

if [ $# -ne 1 ]
then
	echo "Usage: $0 [%cpu]"
	exit
fi

go build -o stop_propaganda cmd/stoppropaganda/main.go
ulimit -n 128000
./stop_propaganda &
sleep 1
stop_pid="$!"
sudo cpulimit --pid "$stop_pid" --limit "$1" &
limit_pid="$!"
echo 'Press Enter to stop dos'
read
sudo kill "$limit_pid"
kill "$stop_pid"
```

**Comment by baylanger** [comment]
The pool is definitely a good approach.

The flag to prevent download content, I'm not sure if this makes more or less impact on the end-point but for sure, it reduces download bandwidth.  Might be worth adding that flag, we can test and decide of a default?!

**Comment by erkexzcx** [comment]
Implemented in v0.0.30
